test/libgit2.jl

```diff
@@ -130,6 +140,47 @@ mktempdir() do dir
             end
         #end
 
+        #@testset "with remote branch and refspec" begin
+            repo = LibGit2.init(cache_repo)
+            try
+                @test isdir(cache_repo)
+                @test isdir(joinpath(cache_repo, ".git"))
+
+                # set a remote branch
+                branch = "upstream2"
+                LibGit2.GitRemote(repo, branch, repo_url,"+refs/heads/$branch:refs/remotes/origin/$branch") |> finalize
+
+                config = joinpath(cache_repo, ".git", "config")
+                lines = split(open(readstring, config, "r"), "\n")
+                @test any(map(x->x == "[remote \"upstream\"]", lines))
+
+                remote = LibGit2.get(LibGit2.GitRemote, repo, branch)
+                @test LibGit2.url(remote) == repo_url
+                @test LibGit2.isattached(repo)
+                finalize(remote)
+            finally
+                finalize(repo)
+            end
+        #end
+
+        #@testset "anonymous" begin
+            repo = LibGit2.init(cache_repo)
+            try
+                @test isdir(cache_repo)
+                @test isdir(joinpath(cache_repo, ".git"))
+
+                # set a remote branch
+                branch = "upstream3"
+                LibGit2.GitRemoteAnon(repo, repo_url) |> finalize
+
+                config = joinpath(cache_repo, ".git", "config")
+                lines = split(open(readstring, config, "r"), "\n")
```

Comment:
I'd want to check that these don't fail if they get windows newlines in them (or check that it's impossible for that to be the case)